build.gradle

```diff
@@ -58,34 +64,58 @@ plugins {
 apply plugin: 'org.grails.grails-web'
 
 apply plugin: 'asset-pipeline'  // enable `assets` block, but see deps below
-apply plugin: 'org.grails.plugins.views-json'  // .gson support
-apply plugin: 'org.owasp.dependencycheck'  // enable dependencyCheckAnalyze task
 
 // prevent warfile issues by applying this after `asset-pipeline`. OBGM-353
 apply plugin: 'org.grails.grails-gsp'
 
+/*
+ * This plugin must load *after* the core grails/gradle plugins.
+ * http://views.grails.org/latest/#_installation
+ */
+apply plugin: 'org.grails.plugins.views-json'
+
+// enable `grails gradle dependencyCheckAnalyze` (security checker)
+apply plugin: 'org.owasp.dependencycheck'
+
 group 'com.openboxes'
 mainClassName = 'org.pih.warehouse.Application'
 sourceCompatibility = 1.8
 targetCompatibility = 1.8
 version '0.9.0-SNAPSHOT'
 
-assetCompile.dependsOn(['npmInstall', 'npm_run_bundle'])
+assetCompile {
+    dependsOn 'npm_run_bundle'
+    /*
+     * Webpack's FileManagerPlugin modifies asset-pipeline's working
+     * directory every time we build, but Gradle's "up-to-date" checks do
+     * not reliably detect those changes. This directive instructs gradle
+     * to always (re)run this task, which only takes a few seconds anyway.
+     */
+    outputs.upToDateWhen { false }
+}
+
+/*
+ * From the asset-pipeline manual: "If you want settings to apply to
+ * both development runtime and build time the properties have to be
+ * duplicated in your application's application.yml [and build.gradle]."
+ *
+ * http://www.asset-pipeline.com/manual/index.html#configuration
+ * http://www.asset-pipeline.com/manual/index.html#configuration-2
+ */
 assets {
-    excludes = [
-        '**/bundle.*.css',
-        '**/bundle.*.js',
-    ]
+    enableDigests = false
+    enableGzip = false
+    maxThreads = 1
+    minifyJs = false  // defer to webpack
+    minifyCss = false  // defer to webpack
     minifyOptions = [
-        languageMode: 'ES6',
-        optimizationLevel: 'SIMPLE',
-        targetLanguage: 'ES5',
+        languageMode: 'ES6'
     ]
 }
 
 bootRun {
     addResources = true
-    dependsOn = ['npmInstall', 'npm_run_bundle']
+    dependsOn 'npm_run_bundle'
```

Comment:
we don't need to depend on both, as `npm_run_bundle` already depends on `npm_install`